### PR TITLE
Normalize totals

### PR DESCRIPTION
**Pull request by berkserbet**
### Why:
* This class normalizes total loads by peak total.

### What:
* Created new normalize class and did not use old one. This function can easily be modified if normalization should be done by peak summer total

### Checklist
- [x ] Finished My Changes
- [ ] Automated Unit Tests

### CODE REVIEW

File: pipelines/rbsa/tasks/normalize_totals.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        return normalization_val
+
+    def validate(self, df):
+        """
+        Validation
+        """
+        logger.info(f'Validating task {self.name}')
+
+        if df.isnull().values.any():
+            logger.exception(f'Task {self.name} did not pass validation. Error found during grouping of sites to zip codes.')
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/normalize_totals.py]: Error message seems to be the same as the site grouping one. Update it to be specific. Something like:
`f'Task {self.name} did not pass validation. Data Frame contains null values when it shouldn't'`

Also, the index_heatcool and the undiscount tasks could use the same update.

File: pipelines/rbsa/tasks/normalize_totals.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            self.did_task_pass_validation = False
+            self.on_failure()
+
+        totals = df[self.enduse_cols].sum(axis=1)
+
+        if round(totals.max(),3) > 1:
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/normalize_totals.py]: add a space after the comma so it's more readable

File: pipelines/rbsa/tasks/normalize_totals.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            logger.exception(f'Task {self.name} did not pass validation. Error found during grouping of sites to zip codes.')
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/normalize_totals.py]: same as the previous logging stmt above

File: pipelines/rbsa/tasks/normalize_totals.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                initialization = False
+            else:
+                normal_loads = normal_loads.append(zipcode_df)
+
+        print(normal_loads[self.enduse_cols].sum(axis=1).max())
+
+        self.validate(normal_loads)
+        self.save_data({self.output_artifact_normal_loads: normal_loads})
+
+
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/normalize_totals.py]: only one new line between function def's

File: pipelines/rbsa/tasks/normalize_totals.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            zipcode_df[self.enduse_cols] = zipcode_df[self.enduse_cols]/normalization_val
+
+            # output dataframe 
+            if initialization:
+                normal_loads = zipcode_df
+                initialization = False
+            else:
+                normal_loads = normal_loads.append(zipcode_df)
+
+        print(normal_loads[self.enduse_cols].sum(axis=1).max())
```
Comment (jongoncalves) [inline pipelines/rbsa/tasks/normalize_totals.py]: logger.info instead print if really necessary to output this

File: pipelines/rbsa/tasks/normalize_totals.py (hunk 1 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/normalize_totals.py]: I'll update all of them

File: pipelines/rbsa/tasks/normalize_totals.py (hunk 2 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/normalize_totals.py]: fixed

File: pipelines/rbsa/tasks/normalize_totals.py (hunk 3 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/normalize_totals.py]: fixed

File: pipelines/rbsa/tasks/normalize_totals.py (hunk 2 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/normalize_totals.py]: fixed

File: pipelines/rbsa/tasks/normalize_totals.py (hunk 4 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/normalize_totals.py]: fixed

File: pipelines/rbsa/tasks/normalize_totals.py (hunk 5 above)
Comment (berkserbet) [reply pipelines/rbsa/tasks/normalize_totals.py]: fixed

**Comment by jongoncalves** [comment]
This can merge in to master 👍

**Pull request merged**